jenius-apps/ambie
# Add support to choose output audio device

This PR adds support for users to choose output device for sound playback.

A new settings card is added to the settings page with a ComboBox listing all currently available devices for playing the sounds.

Users may expect to specifically use a physical or virtual device for Ambie to play sounds. For example, using a sound bar to play Ambie sounds to make people feel cozy in the room while listening to a podcast with a headset. There're also many widely used audio mixer applications like VoiceMeeter, SteelSeries SONAR etc.  SteelSeries's SONAR software adds three virtual audio devices for gaming, media, and chat. Users can easily adjust the balance between those channels or apply different EQ profiles.

### How to test:

1. Launch Ambie and choose a different device as the output device like your monitor, a soundbar, a headset or a virtual device from a mixer software. All the currently playing sounds should be redirected to use the new endpoint for playback in real time.
2. While keeping using the default device, open the system sound settings, change the default output device of the system and check if Ambie immediately uses the new default device to play the sounds.

### Localization:
The strings for Simplified Chinese have been manually translated in this PR. @dpaulino You may need to trigger the machine translation generation again to ensure the strings are also translated in other languages.
